### DISCUSSION TITLE
About the detrending in the Non-Stationary

### DISCUSSION THREAD

**Discussion by ghost** [georgebv/pyextremes#15] [Q&A] [answered]
Hellow

      May I ask whether PYExtremes can only be applied to the stationary model (GEV GPD) and should be de-trended (Time-varying location parameters and time-varying variances) before it is used? If de-trended and stationary model fitting is carried out, how should the trend item be taken into account? How do you add the trend term to the EVA extremum analysis?

Looking forward to your reply, Thanks!

**Comment by georgebv** [comment, marked as answer]
@Goddysen that's a good question. Currently there is no way to analyze non-stationary models. A suggestion was made in another thread https://github.com/georgebv/pyextremes/discussions/11. This would take time to implement and currently I don't have a good idea of how this could be done in a general case - perhaps via a callback function performing pre-processing dynamically. I'll happily accept contributions to the repository as currently I am struggling to find time to finish project documentation.

**Reply by ghost** [reply]
My understanding is that I preprocess the data and remove the trend term. Taking GEV as an example, their three parameters will be constant, but the result of such fitting is that the trend term is not taken into account.So how do you take into account the trend term that you remove, do you add the trend term after the constant parameters that you get from the stationary model?

**Reply by georgebv** [reply]
Correct. You should perform the analysis in the following sequence:

1. Apply transformation to your  signal to remove any periodicity (i.e. de-trending) so that your signal becomes random (as close as possible).
2. Perform EVA on the random signal.
3. Apply inverse transform (add the trend back) to get the actual extreme values values for a given time horizon.

Taking water levels as example, you could remove sea level rise and tide harmonics (2 independent sources or periodicity). The resulting signal would be random. To get extreme value for future you should add projected sea level rise and select relevant tide (depends whether you study high or low extremes) for a given time horizon in the future relative to your baseline (the pivot date against which you de-trend).
